### PR TITLE
Ignore set_time_limit() errors

### PR DESCRIPTION
**Pull request by brandonkelly**
We have a customer whose hosting environment has a custom PHP module installed, which doesn’t allow `set_time_limit(0)` to be called, even though the function exists. Ignoring the error with `@` has proven to work for them.

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

**Comment by samdark** [comment]
That is a bit controversial. Any function could be "checked" like that. We can not add checks everywhere. In addition, there is a standard `disable_functions` and there is already a check for it.

**Comment by samdark** [comment]
Since it's very non-standard environment, I wonder if there is a workaround for it?

File: framework/web/Response.php [hunk 1]
```diff
@@ -431,7 +431,7 @@ protected function sendContent()
         }
 
         if (function_exists('set_time_limit')) {
-            set_time_limit(0); // Reset time limit for big files
+            @set_time_limit(0); // Reset time limit for big files
         } else {
             Yii::warning('set_time_limit() is not available', __METHOD__);
         }
```
Comment (rob006) [inline framework/web/Response.php]: I suggest:

```php
        // try to reset time limit for big files
        if (!@set_time_limit(0)) {
            Yii::warning('set_time_limit() is not available', __METHOD__);
        }
```

**Comment by angrybrad** [comment]
For some context, it's a custom PHP extension called "phplimiter" on a shared hosting environment.

It _does_ allow `set_time_limit` to be called, but it has a range restriction on what the values could be - in this case, between 1-180.

Since `0` is out of that range, a Yii `ErrorException` is raised.

**Comment by samdark** [comment]
@rob006 have you encountered such restrictions as well?

**Comment by rob006** [comment]
> @rob006 have you encountered such restrictions as well?

No, but since we already have check for unavailable `set_time_limit()`, we could extend it to more cases.

File: framework/web/Response.php (hunk 1 above)
Comment (brandonkelly) [reply framework/web/Response.php]: Good point, should be a warning. Done now (07e8a74d219ff6334cf75a6915e18208bcad4337).

**Comment by brandonkelly** [comment]
@rob006 All good here?

**Review by rob006 — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**